llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -4011,15 +4011,17 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
       // Fill in any known bits from range information. There are 3 types being
       // used. The results VT (same vector elt size as BitWidth), the loaded
       // MemoryVT (which may or may not be vector) and the range VTs original
-      // type. The range matadata needs the full range (i.e
+      // type. The range metadata needs the full range (i.e
       // MemoryVT().getSizeInBits()), which is truncated to the correct elt size
       // if it is know. These are then extended to the original VT sizes below.
       if (const MDNode *MD = LD->getRanges()) {
-        computeKnownBitsFromRangeMetadata(*MD, Known0);
+        ConstantInt *Lower = mdconst::extract<ConstantInt>(MD->getOperand(0));
+
+        KnownBits KnownMetadata(Lower->getBitWidth());
+        computeKnownBitsFromRangeMetadata(*MD, KnownMetadata);
+        Known0=KnownMetadata;
```

Comment:
```suggestion
        Known0 = KnownMetadata;
```

Doesn't this need the extension applied?

Reply:
Fixed formatting.  Extend/truncate KnownBits to memory size.